src/main/java/wordsWithSpaces/WordsWithSpaces.java

```diff
@@ -0,0 +1,20 @@
+package wordsWithSpaces;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class WordsWithSpaces {
+    private static Pattern pattern = Pattern.compile("[A-Za-z]\\S*\\b");
+
+    public static String getWords(String text) {
+        Matcher matcher = pattern.matcher(text);
+        List<String> list = new ArrayList<>();
+        while (matcher.find()) {
+            list.add(text.substring(matcher.start(), matcher.end()));
+        }
+        return String.join(" ", list).trim();
```

Comment:
why `trim()` is needed in the end?

Reply:
Because I will have " " as a separator for resulting string and I would like to remove it

Reply:
So it seems not required any more. 
That's value of tests: remove `.trim()` and see if it works :)

Reply:
You were right, it works =D